index.bs

```diff
@@ -990,6 +990,174 @@ while (true) {
 }
 </pre>
 
+## Complete example ##  {#example-complete}
+
+*This section is non-normative.*
+
+This example illustrates use of the closed and ready promises, opening
+of uni-directional and bi-directional streams by either the client or
+the server, and sending and receiving datagrams.
+
+<pre class="example" highlight="js">
+// Adds an entry to the event log on the page, optionally applying a specified
+// CSS class.
+function addToEventLog(text, severity = 'info') {
+  let log = document.getElementById('event-log');
+  let mostRecentEntry = log.lastElementChild;
+  let entry = document.createElement('li');
+  entry.innerText = text;
+  entry.className = 'log-' + severity;
+  log.appendChild(entry);
+
+  // If the most recent entry in the log was visible, scroll the log to the
+  // newly added element.
+  if (mostRecentEntry != null &&
+      mostRecentEntry.getBoundingClientRect().top <
+          log.getBoundingClientRect().bottom) {
+    entry.scrollIntoView();
+  }
+}
+
+// "Connect" button handler.
+async function connect() {
+  let url = document.getElementById('url').value;
+  try {
+    var transport = new WebTransport(url);
+    addToEventLog('Initiating connection...');
+  } catch (e) {
+    addToEventLog('Failed to create connection object. ' + e, 'error');
+    return;
+  }
+
+  try {
+    await transport.ready;
+    addToEventLog('Connection ready.');
+  } catch (e) {
+    addToEventLog('Connection failed. ' + e, 'error');
+    return;
+  }
+
+  transport.closed
+      .then(() => {
+        addToEventLog('Connection closed normally.');
+      })
+      .catch(() => {
+        addToEventLog('Connection closed abruptly.', 'error');
+      });
+
+  globalThis.currentTransport = transport;
```

Comment:
I suggest declaring `currentTransport` with a top-level `let` rather than using `globalThis`. It doesn't make much practical difference, but it's good to promote best practices.

Reply:
Fixed.